types/types.go

```diff
@@ -823,6 +845,27 @@ func (t *StructType) String() string {
 	return b.String()
 }
 
+func (t *StructType) Match(ot Type) bool {
+	if other, ok := ot.(*StructType); ok {
+		switch {
+		case t.Nullability != other.Nullability:
+			return false
+		case t.TypeVariationRef != other.TypeVariationRef:
+			return false
+		}
+		if len(t.Types) != len(other.Types) {
+			return false
+		}
```

Comment:
I would add this to the cases above instead of having it be a separate `if` block

Reply:
with updated PR this won't be needed. So resolving it